smart_ats/jobs/api/permissions.py

```diff
@@ -0,0 +1,10 @@
+from rest_framework import permissions
+from smart_ats.jobs.models import Job
+
+class UpdateOwnJob(permissions.BasePermission):
+    
+    def has_object_permission(self, request, view, obj):
+        
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return obj.id == request.user.id
```

Comment:
you are comparing job id with user id which does not make sense. better compare job_company_admin with user id.
you anyone with company level access to be able to update the job.